test/handler_options_test.exs

```diff
@@ -88,5 +88,79 @@ defmodule ExVCR.Adapter.HandlerOptionsTest do
         HttpServer.stop(@port)
       end
     end
+
+    defp always_map(headers) do
+      # When a request is first recorded then the request headers are stored
+      # as a List, but when it's fetched from storage then they are a Map...
+      if(is_list(headers), do: Enum.into(headers, %{}), else: headers)
```

Comment:
I'm not sure if I missed something with this, but maybe on a different issue we can make this consistent (either a `List` or a `Map`).